File: Programowanie_Obiektowe/PO_JAVA/src/Czlowiek.java
public class Czlowiek extends Zwierze
{
    public Czlowiek(int x, int y, Swiat swiat)
    {
        super(Stale.CZLOWIEK_SILA, Stale.CZLOWIEK_INICJATYWA, x, y, swiat);
        this.umiejetnosc = -(Stale.CZAS_TRWANIA_UMIEJETNOSCI + Stale.CZAS_REGENERACJI_UMIEJETNOSCI);
        this.tmp_sila = 0;
    }

    @Override
    public void akcja()
    {
        if (umiejetnosc == wiek)        //umiejetnosc aktywowana
        {
            tmp_sila = sila;
            sila = Stale.BOOST_SILY;
        }
        else if (Stale.CZAS_TRWANIA_UMIEJETNOSCI + umiejetnosc - wiek > 0)      //umiejetnosc aktywna
        {
            sila--;
        }
        else if (Stale.CZAS_TRWANIA_UMIEJETNOSCI + umiejetnosc - wiek == 0)     //umiejentonosc wygasla
        {
            sila = sila - (Stale.BOOST_SILY - Stale.CZAS_TRWANIA_UMIEJETNOSCI + 1) + tmp_sila;
        }

        wiek++;
        przesun(gx, gy);
    }

    @Override
    public void rozmnazanie()
    {
        // Człowiek się nie rozmnaża
    }

    @Override
    public char rysowanie()
    {
        return Stale.CZLOWIEK;
    }

    @Override
    public String nazwa()
    {
        return Stale.CZLOWIEK_S;
    }

    @Override
    public Organizm stworzPotomka(int x, int y)
    {
        return null;
    }
}

File: Programowanie_Obiektowe/PO_C/PO_C/Czlowiek.cpp
#include "Czlowiek.h"

Czlowiek::Czlowiek(int x, int y, Swiat& swiat) : Zwierze(CZLOWIEK_SILA, CZLOWIEK_INICJATYWA, x, y, swiat) 
{
    umiejetnosc = -(CZAS_TRWANIA_UMIEJETNOSCI + CZAS_REGENERACJI_UMIEJETNOSCI);
    tmp_sila = 0;
}

void Czlowiek::akcja() 
{
    if (umiejetnosc == wiek)                //umiejetnosc aktywowana
    {
        tmp_sila = sila;
        sila = BOOST_SILY;
    }
    else if (CZAS_TRWANIA_UMIEJETNOSCI + umiejetnosc - wiek > 0)    //umiejetnosc aktywna
    {
        sila--;
    }
    else if (CZAS_TRWANIA_UMIEJETNOSCI + umiejetnosc - wiek == 0)   //umiejentonosc wygasla
    {
        sila = sila - (BOOST_SILY - CZAS_TRWANIA_UMIEJETNOSCI + 1) + tmp_sila;         //bierzemy pod uwage inne
    }

    wiek++;

    przesun(gx, gy);
}

void Czlowiek::rozmnazanie()
{
    return;
}

char Czlowiek::rysowanie() const 
{
    return CZLOWIEK;
}

Organizm* Czlowiek::stworzPotomka(int x, int y) const 
{
    return nullptr;
}